### PR TITLE
Maintenance updates for corepack, pnpm v8 and other minor housekeeping

### PR DESCRIPTION
**Pull request by simonihmig**
Wanted to update some dependencies (PR comming), but things were not working locally for me, so did some general maintenance here:
* use corepack
* switch to pnpm v8
* update some Embroider dependencies

The only CI failure is the Cloudflare deployment, and I'M not sure whatÄs happening there. I don't think I have access to their logs there...

### CODE REVIEW

File: .github/actions/pnpm/action.yml [hunk 1]
```diff
@@ -3,12 +3,10 @@ description: Setup node and install dependencies using pnpm
 runs:
   using: "composite"
   steps:
-    - uses: pnpm/action-setup@v2.2.2
+    - uses: pnpm/action-setup@v4
+    - uses: actions/setup-node@v4
       with:
-        version: 7
```
Comment (simonihmig) [inline .github/actions/pnpm/action.yml:8]: the updated action understands to look into `package.json#packageManager` to use the version defined there!

**Review by ynotdraw — APPROVED** [review]
@simonihmig I've got access to Cloudflare - here is what I'm seeing from the logs:

```bash
06:38:53.596 | ERROR: This version of pnpm requires at least Node.js v18.12
-- | --
06:38:53.597 | The current version of Node.js is v16.20.2
06:38:53.597 | Visit https://r.pnpm.io/comp to see the list of past pnpm versions with respective Node.js version support.
```

I see you updated `engines` to be a newer version of `node`, but I'm wondering if we need to tell CI which version of node to use as well to ensure it's the version expected?

**Review by nicolechung — APPROVED** [review]

**Comment by nicolechung** [comment]
Thank you @simonihmig  for updating this package!

**Comment by simonihmig** [comment]
Thanks @ynotdraw, will look into this! Wasn't sure if that error is actually related to my changes or it just began to fail some time ago, but seems it is (related)!  😕

**Comment by ynotdraw** [comment]
> Thanks @ynotdraw, will look into this! Wasn't sure if that error is actually related to my changes or it just began to fail some time ago, but seems it is (related)! 😕

No problem! Updated the node version and pnpm version in Cloudflare, we're all green now.  :shipit: 🎉

**Review by kellyselden — APPROVED** [review]

**Pull request merged**